homeassistant/components/irm_kmi/__init__.py

```diff
@@ -0,0 +1,64 @@
+"""Integration for IRM KMI weather."""
+
+import logging
+
+from irm_kmi_api import IrmKmiApiClientHa
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, IRM_KMI_TO_HA_CONDITION_MAP, PLATFORMS, USER_AGENT
+from .coordinator import IrmKmiCoordinator
+from .data import IrmKmiConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: IrmKmiConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+
+    entry.runtime_data = IrmKmiApiClientHa(
+        session=async_get_clientsession(hass),
+        user_agent=USER_AGENT,
+        cdt_map=IRM_KMI_TO_HA_CONDITION_MAP,
+    )
+
+    hass.data[DOMAIN][entry.entry_id] = irm_kmi_coordinator = IrmKmiCoordinator(
```

Comment:
If possible, make a dedicated class where you keep the client and the coordinator. Then use the dedicated class and move that completely in runtime data.

Reply:
I made a class as suggested.  However, I had to also give the API client to the IrmKmiCoordinator and not just put it in the ConfigEntry or I got circular dependencies.